### ISSUE TITLE
False negative ExplicitJobPermissions when both workflow and job permissions are defined

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#277]
### Version

ghlint 0.5.0

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
    ```yaml
    name: "Onboard GitHub repository to Renovate"
    
    on: push
    permissions:
      pull-requests: write
      contents: write
    jobs:
      job:
        runs-on: ubuntu-latest
        permissions:
          pull-requests: write
          contents: write
        steps:
          - uses: actions/checkout@v4
    ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

ExplicitJobPermissions because the workflow ones are redundant.

### Actual behavior

Nothing.

### Additional Context

ref: platform-52
